appservice/src/main/java/nu/yona/server/analysis/rest/ActivityControllerBase.java

```diff
@@ -40,136 +32,115 @@
 import nu.yona.server.analysis.service.WeekActivityDTO;
 import nu.yona.server.analysis.service.WeekActivityOverviewDTO;
 import nu.yona.server.crypto.CryptoSession;
-import nu.yona.server.goals.rest.GoalController;
 import nu.yona.server.subscriptions.service.UserService;
 
 /*
- * Controller to retrieve activity data for a user.
+ * Activity controller base class.
  */
-@Controller
-@RequestMapping(value = "/users/{userID}/activity", produces = { MediaType.APPLICATION_JSON_VALUE })
-public class ActivityController
+abstract class ActivityControllerBase
 {
 	@Autowired
-	private ActivityService activityService;
+	protected ActivityService activityService;
 
 	@Autowired
 	private UserService userService;
 
 	@Autowired
 	private CurieProvider curieProvider;
 
-	@RequestMapping(value = "/weeks/", method = RequestMethod.GET)
-	@ResponseBody
-	public HttpEntity<PagedResources<WeekActivityOverviewResource>> getWeekActivityOverviews(
-			@RequestHeader(value = PASSWORD_HEADER) Optional<String> password, @PathVariable UUID userID, Pageable pageable,
-			PagedResourcesAssembler<WeekActivityOverviewDTO> pagedResourcesAssembler)
+	protected final String WEEK_ACTIVITY_OVERVIEWS_URI_FRAGMENT = "/weeks/";
+	protected final String DAY_OVERVIEWS_URI_FRAGMENT = "/days/";
+	protected final String WEEK_ACTIVITY_DETAIL_URI_FRAGMENT = "/weeks/{date}/details/{goalID}";
+	protected final String DAY_ACTIVITY_DETAIL_URI_FRAGMENT = "/days/{date}/details/{goalID}";
+	protected final String GOAL_PATH_VARIABLE = "goalID";
+	protected final String DATE_PATH_VARIABLE = "date";
+
+	protected HttpEntity<PagedResources<WeekActivityOverviewResource>> getWeekActivityOverviews(Optional<String> password,
+			UUID userID, Pageable pageable, PagedResourcesAssembler<WeekActivityOverviewDTO> pagedResourcesAssembler,
+			Supplier<Page<WeekActivityOverviewDTO>> activityFunction, LinkProvider linkProvider)
```

Comment:
Rename ``activityFunction`` into ``activitySupplier``, as the type as well as the functionality indicate that.